feat(manager/flux): Support system manifests outside `flux-system` subdirectory (#22966)
renovatebot/renovate: lib/modules/manager/flux/readme.md

### Flux system manifests support

> Renovate supports updating Flux system manifests generated during [Flux installation](https://fluxcd.io/docs/installation/#customize-flux-manifests). System manifests are identified via being named `gotk-components.y(a)ml` and containing comment headers as generated by `flux bootstrap`. Additional components are parsed and provided to `flux install` to generate a new manifest file.

Reviewer: ```suggestion
Renovate supports updating Flux system manifests generated during [Flux installation](https://fluxcd.io/docs/installation/#customize-flux-manifests).
System manifests are identified via being named `gotk-components.y(a)ml` and containing comment headers as generated by `flux bootstrap`.
Additional components are parsed and provided to `flux install` to generate a new manifest file.
```
one sentence per line

Reviewer 2: ```suggestion
Renovate can update Flux system manifests generated during [Flux installation](https://fluxcd.io/docs/installation/#customize-flux-manifests).
Flux system manifests files are `gotk-components.yml` or `gotk-components.yaml`.
These files also have comment headers generated by the `flux bootstrap` command.
Additional components are parsed and provided to `flux install` to generate a new manifest file.
```

I think this is better? This line still feels vague:

> Additional components are parsed and provided to `flux install` to generate a new manifest file.

We don't explain _which_ components are parsed and provided to `flux install`. If we add that explanation we probably make the text too verbose and in-depth. We should focus on the Renovate-specific parts, I think. So maybe it's best to drop that last line.

Author: I agree, it was leaning a little too much into implementation details. I've applied the suggested changes and removed the latter line about components.

On that though, is it worth popping a warning about [custom flags not being supported](https://github.com/renovatebot/renovate/issues/13952#issuecomment-1031620639) for things like `--cluster-name`? In such a case Renovate will overwrite these values with the defaults in its PR.

Updating system manifests requires that either:

1. The `flux` tool is pre-installed, or
1. You run a Docker image based on [containerbase](https://github.com/containerbase), such as the official Renovate images, and have `binarySource=install` configured

### Default fileMatch

> By default, the `flux` manager will only match `**/gotk-components.y(a)ml` (i.e. system manifest) files from the bootstrap process.

Reviewer 2: ```suggestion
By default, the `flux` manager only matches `**/gotk-components.y(a)ml` .
So it only checks system manifest files generated by the `flux bootstrap` command.
```

How about something like this?

### Flux system manifests support

Renovate supports updating Flux system manifests generated during [Flux installation](https://fluxcd.io/docs/installation/#customize-flux-manifests).
System manifests are identified via being named `gotk-components.y(a)ml` and containing comment headers as generated by `flux bootstrap`.

> 

Reviewer 2: Just a comment to say that you should update your branch to match the current `main`, because we just merged in a bunch of `v36` breaking changes. 😉

Author: Aah ok, I'll merge in `main`

Updating system manifests requires that either:

1. The `flux` tool is pre-installed, or
1. You run a Docker image based on [containerbase](https://github.com/containerbase), such as the official Renovate images, and have `binarySource=install` configured

> 

Reviewer 2: > I agree, it was leaning a little too much into implementation details. I've applied the suggested changes and removed the latter line about components.
> 
> On that thought, is it worth popping a warning about [custom flags not being supported](https://github.com/renovatebot/renovate/issues/13952#issuecomment-1031620639) for things like `--cluster-name`? In such a case Renovate will overwrite these values with the defaults in its PR.

Overwriting a users custom change is something that we should probably warn about. Here's the _code_ for a Material for MkDocs admonition. The `<!-- prettier-ignore -->` is needed to keep Prettier away. By default Prettier removes the 4 whitespaces we need for the admonition block. 😄 

```markdown
<!-- prettier-ignore -->
!!! warning
    Four blank spaces so the text lines up with the warning word.
    Add four blank spaces if you need another sentence.
```

I'll let you put the admonition warning in a logical place. 😉

Author: Ok, sounds good to me. I've added a warning and linked the issue.
